distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/cli/action/InstallPluginAction.java

```diff
@@ -271,24 +283,37 @@ private static void handleInstallXPack(final Build.Flavor flavor) throws UserExc
     private Path download(PluginDescriptor plugin, Path tmpDir) throws Exception {
         final String pluginId = plugin.getId();
 
-        if (OFFICIAL_PLUGINS.contains(pluginId)) {
+        final String logPrefix = terminal.isHeadless() ? "" : "-> ";
+
+        // See `InstallPluginCommand` it has to use a string argument for both the ID and the location
+        if (OFFICIAL_PLUGINS.contains(pluginId) && (plugin.getLocation() == null || plugin.getLocation().equals(pluginId))) {
+            final String pluginArchiveDir = System.getenv("ES_PLUGIN_ARCHIVE_DIR");
```

Comment:
Cloud wanted to be able to install plugins in exactly the same way between the Cloud and Cloud ESS Docker images. Currently in `master`, we do this with the `plugin-wrapper.sh` script, which intercepts install commands and uses a cached zip if the script finds one in the archive directory. That approach doesn't work in the declarative model, so the Java installer code does this interception instead.

I suppose I could hard-code the archive path instead of using `ES_PLUGIN_ARCHIVE_DIR`, but that felt a bit more obscure.

Reply:
> I suppose I could hard-code the archive path instead of using ES_PLUGIN_ARCHIVE_DIR, but that felt a bit more obscure.

I think the alternative here is to provide a location to the path inside the container where the archive is in the config file vs _only_ supplying an id, and then having some other logic resolve the id to the local archive. I don't have strong feelings either way other than the `location` based approach being slightly less opaque and perhaps flexible.